tests/unit-tests/test-class-lesson.php

```diff
@@ -101,13 +100,110 @@ public function testIsPreRequisiteComplete() {
+			$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+			$this->assertEquals(
+				$expected_course_id,
+				$course_id,
+				"Lesson with ID {$lesson_id} has course ID {$course_id}, expected {$expected_course_id}"
+			);
+		}
+	}
+
+	public function testGetCourseIds() {
+		$this->assertTrue(
+			method_exists( 'WooThemes_Sensei_Lesson', 'get_course_ids' ),
+			'The lesson class method `get_course_ids` does not exist '
+		);
+		$course_ids             = $this->factory->course->create_many( 3 );
+		$lesson_ids             = $this->factory->lesson->create_many( 9 );
+		$lesson_id_to_course_id = array();
+		foreach ( $lesson_ids as $lesson_id ) {
+			$course_index                         = array_rand( $course_ids );
+			$course_id                            = $course_ids[ $course_index ];
+			$lesson_id_to_course_id[ $lesson_id ] = $course_id;
+			update_post_meta( $lesson_id, '_lesson_course', $course_id );
+		}
+		$courses_id = Sensei()->lesson->get_course_ids( $lesson_ids );
+		foreach ( $lesson_id_to_course_id as $lesson_id => $expected_course_id ) {
+			$course_id            = $courses_id[ $lesson_id ];
+			$get_course_id_result = Sensei()->lesson->get_course_id( $lesson_id );
+			$this->assertEquals(
+				$expected_course_id,
+				$course_id,
+				"Lesson with ID {$lesson_id} has course ID {$course_id}, expected {$expected_course_id}"
+			);
+			$this->assertEquals(
+				$get_course_id_result,
+				$course_id,
+				"get_course_ids returned ID {$course_id} for lesson {$lesson_id}, but get_course_id returned {$get_course_id_result}"
+			);
+		}
+	}
+
+
+	public function testGetCourseIdsCache() {
+		$this->assertTrue(
+			method_exists( 'WooThemes_Sensei_Lesson', 'get_course_ids' ),
+			'The lesson class method `get_course_ids` does not exist '
+		);
+		$course_ids             = $this->factory->course->create_many( 3 );
+		$lesson_ids             = $this->factory->lesson->create_many( 9 );
+		$lesson_id_to_course_id = array();
+		foreach ( $lesson_ids as $lesson_id ) {
+			$course_index                         = array_rand( $course_ids );
+			$course_id                            = $course_ids[ $course_index ];
+			$lesson_id_to_course_id[ $lesson_id ] = $course_id;
+			update_post_meta( $lesson_id, '_lesson_course', $course_id );
+		}
+		$courses_id = Sensei()->lesson->get_course_ids( $lesson_ids );
+		foreach ( $lesson_id_to_course_id as $lesson_id => $expected_course_id ) {
+			$course_id            = $courses_id[ $lesson_id ];
+			$get_course_id_result = Sensei()->lesson->get_course_id( $lesson_id );
+			$this->assertEquals(
+				$expected_course_id,
+				$course_id,
+				"Lesson with ID {$lesson_id} has course ID {$course_id}, expected {$expected_course_id}"
+			);
+			$this->assertEquals(
+				$get_course_id_result,
+				$course_id,
+				"get_course_ids returned ID {$course_id} for lesson {$lesson_id}, but get_course_id returned {$get_course_id_result}"
+			);
+		}
+		shuffle( $lesson_ids );
```

Comment:
And this test is the same as the previous one until this line, right? Should we extract this part to a private method here?

Reply:
`testGetCourseIds` and `testGetCourseIdsCache` are basically the same, yeah.

I'm not sure about extracting this part to a private method, tho, because that means basically moving almost all of the `testGetCourseIds`, **including** the assertions, and just returning two variables: `$lesson_ids` and `$lesson_id_to_course_id`, as we need both of these variables to check against the cache.

 I think it could make sense to delete the `testGetCourseIds` and rename `testGetCourseIdsCache` to `testGetCourseIds`, but..I'm not sure. WDYT?

Reply:
Done in [d0963e5](https://github.com/Automattic/sensei/pull/5294/commits/d0963e517d031e03a26b97966f612fa48188cfa8).